src/protocol/primitives/uvarint.spec.ts

```diff
@@ -0,0 +1,46 @@
+import { ReadBuffer, WriteBuffer } from '../serialization';
+import { UVarInt } from './uvarint';
+
+describe('UVarInt', () => {
+  const cases = [
+    { value: UVarInt.MIN_VALUE, buffer: Buffer.from([0x00]) },
+    { value: 1, buffer: Buffer.from([0x01]) },
+    { value: 127, buffer: Buffer.from([0x7f]) },
+    { value: 128, buffer: Buffer.from([0x80, 0x01]) },
+    { value: 255, buffer: Buffer.from([0xff, 0x01]) },
+    { value: 300, buffer: Buffer.from([0xac, 0x02]) },
+    { value: 16384, buffer: Buffer.from([0x80, 0x80, 0x01]) },
+    { value: UVarInt.MAX_VALUE, buffer: Buffer.from([0xff, 0xff, 0xff, 0xff, 0x0f]) }
```

Comment:
Please avoid referring to the implementation details in the test scenario. If the `MAX_VALUE` is changed in the implementation, the behavior of this test scenario will change too.

Reply:
For me, it's not an implementation detail but something that is clearly defined in the protocol.

Reply:
The value is part of the specification, but the constant declaration is nothing more but the developers choice how to implement this. IMO the test should check against the specified value, not agains the declared constant.

Reply:
By having this test like this we prevent changing the value like that. It's something like having MAX_SAFE_INTEGER on Number, it should never change as it is a constant part of the specification. And if it changes by any chance I want to have a closer look